src/inverters/dto/custom-entity.ts

```diff
@@ -35,19 +35,34 @@ export interface CustomEntity extends HassEntity {
      * @param invert
      */
     toPower(invert?: boolean): number;
+
+    /**
+     * Auto converts the state to watts/kilowatts, with the suffix
+     * @param invert
+     * @param decimals
+     * @param scale
+     */
+    toPowerString(scale?: boolean, decimals?: number, invert?: boolean): string;
+
+    getUOM(): UnitOfPower | UnitOfEnergy | UnitOfElectricalCurrent
 }
 
 // Function to convert HassEntity to CustomEntity
 export function convertToCustomEntity(entity: any): CustomEntity {
     return {
         ...entity,
         toNum: (decimals?: number, invert?: boolean) => Utils.toNum(entity?.state, decimals, invert),
-        isValid: () => entity?.state !== null && entity.state !== undefined || false,
+        isValid: () => entity?.state !== null && entity.state !== undefined && entity.state !== 'unknown' || false,
         notEmpty: () => entity?.state !== '' || false,
         isNaN: () => Number.isNaN(entity?.state) || true,
         toPower: (invert?: boolean) => (entity.attributes?.unit_of_measurement || '').toLowerCase() === 'kw'
             ? Utils.toNum(((entity?.state || '0') * 1000), 0, invert)
             : Utils.toNum((entity?.state || '0'), 0, invert) || 0,
-        toString: () => entity?.state?.toString() || ''
+        toPowerString: (scale?: boolean, decimals?: number, invert?: boolean) =>
+            scale ?
+                Utils.convertValueNew(entity?.state, entity?.attributes?.unit_of_measurement, decimals || 1) :
```

Comment:
This does not appear to be working properly. When scale is true it does not auto convert the values. Tested on pv1 and pv2 power

Reply:
I've updated the function to include power values